test/units/modules/network/nxos/test_nxos_vlans.py

```diff
@@ -0,0 +1,138 @@
+
+        self.mock_get_resource_connection_config = patch('ansible.module_utils.network.common.cfg.base.get_resource_connection')
+        self.get_resource_connection_config = self.mock_get_resource_connection_config.start()
+
+        self.mock_get_resource_connection_facts = patch('ansible.module_utils.network.common.facts.facts.get_resource_connection')
+        self.get_resource_connection_facts = self.mock_get_resource_connection_facts.start()
+
+        self.mock_edit_config = patch('ansible.module_utils.network.nxos.config.vlans.vlans.Vlans.edit_config')
+        self.edit_config = self.mock_edit_config.start()
+
+        self.mock_get_device_data = patch('ansible.module_utils.network.nxos.facts.vlans.vlans.VlansFacts.get_device_data')
+        self.get_device_data = self.mock_get_device_data.start()
+
+    def tearDown(self):
+        super(TestNxosVlansModule, self).tearDown()
+        self.mock_FACT_LEGACY_SUBSETS.stop()
+        self.mock_get_resource_connection_config.stop()
+        self.mock_get_resource_connection_facts.stop()
+        self.mock_edit_config.stop()
+
+    def load_fixtures(self, commands=None, device=''):
+        self.mock_FACT_LEGACY_SUBSETS.return_value = dict()
+        self.edit_config.return_value = None
+
+        def load_from_file(*args, **kwargs):
+            cmd = args[1]
+            filename = str(cmd).split(' | ')[0].replace(' ', '_')
+            return load_fixture('nxos_vlans', filename)
+        self.get_device_data.side_effect = load_from_file
+
+    def test_1(self):
+        '''
+        vlan 1,3-5
+        vlan 3
+          name test-vlan3
+        !Note:vlan 4 is present with default settings
+        vlan 5
+          shutdown
+          name test-changeme
+          mode fabricpath
+          state suspend
+          vn-segment 942
+        !Note:vlan 7 is not present
+        vlan 8
+          shutdown
+        '''
+        playbook = dict(config=[
+            dict(vlan_id=4),
+            dict(vlan_id=5, mapped_vni=555, mode='ce'),
+            dict(vlan_id=7, mapped_vni=777, name='test-vlan7', enabled=False),
+            dict(vlan_id=8)
+            # vlan 3 is not present in playbook.
+        ])
+
+        merged = [
+            # Update existing device states with any differences in the playbook.
+            'vlan 5', 'vn-segment 555', 'mode ce',
+            'vlan 7', 'vn-segment 777', 'name test-vlan7', 'shutdown'
+        ]
+        playbook['state'] = 'merged'
+        set_module_args(playbook, ignore_provider_arg)
+        self.execute_module(changed=True, commands=merged)
+
+        deleted = [
+            # Reset existing device state to default values. Scope is limited to
+            # objects in the play.For vlans this means deleting each vlan listed
+            # in the playbook. Ignores any play attrs other than 'vlan_id'.
+            'no vlan 4',
+            'no vlan 5',
+            'no vlan 8'
+        ]
+        playbook['state'] = 'deleted'
```

Comment:
How does deleted work for this module?  Does it only delete vlans that are specified in the playbook or does it delete all vlans?

Reply:
When `config` is specified it removes only the vlans specified by config. When not specified it removes all vlans. I'll include a test for that too.